### PR TITLE
Add templatised version of AlignSizeUp

### PR DESCRIPTION
**Pull request by vamsimanchala**
Implement a template version of AlignSizeUp that returns the increased size as a multiple of alignment.
BUG=http://b/208703041

### CODE REVIEW

File: tensorflow/lite/micro/memory_helpers_test.cc [hunk 1]
```diff
@@ -31,6 +31,12 @@ void* FakeAllocatePersistentBuffer(TfLiteContext* context, size_t bytes) {
   return reinterpret_cast<void*>(global_persistent_buffer);
 }
 
+// Test structure to test AlignSizeUp.
+struct alignas(32) TestAlignSizeUp {
+  // Opaque blob
+  float blob_data[4];
+};
```
Comment (advaitjain) [inline tensorflow/lite/micro/memory_helpers_test.cc:38]: let's move this struct inside the test case.

File: tensorflow/lite/micro/memory_helpers_test.cc [hunk 2]
```diff
@@ -111,6 +117,11 @@ TF_LITE_MICRO_TEST(TestAlignSizeUp) {
   TF_LITE_MICRO_EXPECT_EQ(static_cast<size_t>(24), tflite::AlignSizeUp(23, 2));
   TF_LITE_MICRO_EXPECT_EQ(static_cast<size_t>(24), tflite::AlignSizeUp(23, 3));
   TF_LITE_MICRO_EXPECT_EQ(static_cast<size_t>(32), tflite::AlignSizeUp(23, 16));
+
+  TF_LITE_MICRO_EXPECT_EQ(static_cast<size_t>(32),
+                          tflite::AlignSizeUp<TestAlignSizeUp>());
+  TF_LITE_MICRO_EXPECT_EQ(static_cast<size_t>(64),
+                          tflite::AlignSizeUp<TestAlignSizeUp>(2));
```
Comment (advaitjain) [inline tensorflow/lite/micro/memory_helpers_test.cc:124]: Let's have a separate test case for the templatized align size up.

**Review by advaitjain — CHANGES_REQUESTED** [review]
nice! Only nit comments.

**Comment by advaitjain** [comment]
I have updated BUG= in the PR description to http://b/208703041 from b/208703041

File: tensorflow/lite/micro/memory_helpers_test.cc (hunk 1 above)
Comment (vamsimanchala) [reply tensorflow/lite/micro/memory_helpers_test.cc]: Done

File: tensorflow/lite/micro/memory_helpers_test.cc (hunk 2 above)
Comment (vamsimanchala) [reply tensorflow/lite/micro/memory_helpers_test.cc:126]: Done

**Review by advaitjain — APPROVED** [review]

**Pull request merged**